src/dosext/mfs/mfs.c

```diff
@@ -3996,13 +4026,15 @@ static int dos_fs_redirect(struct vm86_regs *state)
       int return_val;
       f = &open_files[sft_fd(sft)];
 
-      cnt = WORD(state->ecx);
       if (f->name == NULL) {
         SETWORD(&state->eax, ACCESS_DENIED);
         return FALSE;
       }
+
+      update_seek_from_dos(sft_position(sft), &f->seek);
+      cnt = WORD(state->ecx);
       if (cnt) {
-        int cnt1 = region_lock_offs(f->fd, sft_position(sft), cnt);
+        int cnt1 = region_lock_offs(f->fd, f->seek, cnt);
```

Comment:
Now this won't work, as we use some
special 64bit locking offsets to emulate
share modes. :( See open_compat() & co.
So I would suggest to just not check
the lock if we are above 32bits.